x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -65,6 +65,18 @@ export type EventCategoryOverride = t.TypeOf<typeof event_category_override>;
 export const eventCategoryOverrideOrUndefined = t.union([event_category_override, t.undefined]);
 export type EventCategoryOverrideOrUndefined = t.TypeOf<typeof eventCategoryOverrideOrUndefined>;
 
+export const tiebreaker_field = t.string;
+export type TiebreakerField = t.TypeOf<typeof tiebreaker_field>;
```

Comment:
Just a note, @banderror pointed out over in https://github.com/elastic/kibana/pull/132409 that if the `const` declaration is the same name as the type the JSDoc's will transfer over as well, which is a nice little usability perk in the IDE. You can see an example over in the new rule_params:

https://github.com/elastic/kibana/blob/fb1eeb0945e25928a2516133055f048ff098166f/x-pack/plugins/security_solution/common/detection_engine/schemas/common/rule_params.ts#L46-L57

Reply:
@e40pud I was gonna create a ticket for addressing issues with the rule schemas we have, and we're considering adding this to our 8.4 plan for the rules area (yet to be decided though). As part of that, I can create one for cleaning up `OrUndefined` stuff.

Reply:
@e40pud @spong it took a while, but here we are:

- https://github.com/elastic/kibana/issues/138614
- https://github.com/elastic/kibana/issues/138606